src/components/Accordion/Accordion.js

```diff
@@ -39,19 +39,60 @@ const getComponentClassName = ({
   )
 }
 
-class Accordion extends Component<AccordionProps, AccordionState> {
+const buildOpenSections = sectionIds =>
+  sectionIds.reduce(
+    (accumulator, id) => ({
+      ...accumulator,
+      [id]: true,
+    }),
+    {}
+  )
+
+class Accordion extends PureComponent<AccordionProps, AccordionState> {
   static Body = Body
   static Section = Section
   static Title = Title
 
   static defaultProps = {
     onOpen: noop,
     onClose: noop,
+    openSectionIds: [],
     size: 'md',
   }
 
-  state = {
-    sections: {},
+  constructor(props) {
+    super(props)
+
+    this.state = {

```

Comment:
@knicklabs I believe you can just write:

```js
state = {
  sections: buildOpenSections(this.props.openSectionIds)
}
```

No constructor needed. The babel transformer should take care of it :)
I only learned about this like 2 months ago

Reply:
I originally had something like that, but changed it when I started using the props to the set the initial state. Wasn't aware we could use `this.props` there.